src/vfs.js

```diff
@@ -91,14 +94,24 @@ export class VFS {
 
 		name = relative(this.rootDirectory, filename);
 
-		if (filename.endsWith(".server.lua")) {
+		let serverMatch = /^([^\.]+)\.server\.lua$/.exec(name);
+		let clientMatch = /^([^\.]+)\.client\.lua$/.exec(name);
+		let moduleMatch = /^([^\.]+)\.lua$/.exec(name);
+
+		if (serverMatch) {
 			type = "Script";
-			name = name.replace(/\.server\.lua$/, "");
-		} else if (filename.endsWith(".client.lua")) {
+			name = serverMatch[1];
+		}
+		else if (clientMatch) {
 			type = "LocalScript";
-			name = name.replace(/\.client\.lua$/, "");
-		} else {
-			name = name.replace(/\.lua$/, "");
+			name = clientMatch[1];
+		}
+		else if (moduleMatch) {
+			type = "ModuleScript";
+			name = moduleMatch[1];
+		}
+		else {
```

Comment:
Style nitpick: We should put the `else` branches on the same line as the closing braces